pkg/apis/application/v1alpha1/types.go

```diff
@@ -437,6 +439,111 @@ type SyncPolicyAutomated struct {
 	Prune bool `json:"prune,omitempty" protobuf:"bytes,1,opt,name=prune"`
 	// SelfHeal enables auto-syncing if  (default: false)
 	SelfHeal bool `json:"selfHeal,omitempty" protobuf:"bytes,2,opt,name=selfHeal"`
+	// MaintenanceWindows disables auto-syncing during the configured time windows
+	MaintenanceWindows MaintenanceWindows `json:"maintenanceWindows,omitempty" protobuf:"bytes,3,opt,name=maintenanceWindows"`
+}
+
+func (s *SyncPolicy) IsAutomated() bool {
+	return s != nil && s.Automated != nil
+}
+
+type MaintenanceWindows []*MaintenanceWindow
+
+// MaintenanceWindow controls when automated syncs should be disabled
+type MaintenanceWindow struct {
+	// Schedule is the time the window will begin, specified in cron format
+	Schedule string `json:"schedule,omitempty" protobuf:"bytes,1,opt,name=schedule"`
+	// Duration is the amount of time the maintenance window will be open
+	Duration string `json:"duration,omitempty" protobuf:"bytes,2,opt,name=duration"`
+}
+
+func (m *MaintenanceWindows) IsZero() bool {
+	return len(*m) == 0
```

Comment:
do you need the `*m` would just `m` work

Reply:
I tried it and it did not work.